src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +855,210 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
+	removeSet->setVariablePlug()->setValue( "setMembership:set" );
+
+	auto rowsPlug = static_cast<Spreadsheet::RowsPlug *>(
+		PlugAlgo::promoteWithName( spreadsheet->rowsPlug(), "edits" )
+	);
+
+	Metadata::registerValue( rowsPlug, "spreadsheet:defaultRowVisible", new BoolData( false ) );
+	Metadata::registerValue( rowsPlug->defaultRow(), "spreadsheet:rowNameWidth", new IntData( 300 ) );
+
+	result->outPlug()->setInput( removeSet->outPlug() );
+
+	return result;
+}
+
+EditScope::ProcessorRegistration g_setMembershipProcessorRegistration( "SetMembershipEdits", setMembershipProcessor );
+
+}  // namespace
+
+void EditScopeAlgo::setSetMembership( Gaffer::EditScope *scope, const std::string &set, const IECore::PathMatcher &paths, EditScopeAlgo::SetMembership state )
+{
+	Node *processor = scope->acquireProcessor( "SetMembershipEdits" );
+	auto rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+
+	auto row = rows->row( set );
+	if( !row )
+	{
+		row = rows->addRow();
+		row->namePlug()->setValue( set );
+	}
+
+	auto addCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_addSetColumnIndex );
+	auto removeCell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( g_removeSetColumnIndex );
+
+	auto addStringPlug = addCell->valuePlug<StringVectorDataPlug>();
+	auto removeStringPlug = removeCell->valuePlug<StringVectorDataPlug>();
+
+	ConstStringVectorDataPtr addData = addStringPlug->getValue();
+	ConstStringVectorDataPtr removeData = removeStringPlug->getValue();
+
+	PathMatcher addPaths( addData->readable().begin(), addData->readable().end() );
+	PathMatcher removePaths( removeData->readable().begin(), removeData->readable().end() );
+
+	if( state == EditScopeAlgo::SetMembership::INHERIT )
+	{
+		if( !addPaths.removePaths( paths ) )
+		{
+			removePaths.removePaths( paths );
+		}
+	}
+	else if( state == EditScopeAlgo::SetMembership::IN )
+	{
+		addPaths.addPaths( paths );
+		removePaths.removePaths( paths );
+	}
+	else if( state == EditScopeAlgo::SetMembership::OUT )
+	{
+		addPaths.removePaths( paths );
+		removePaths.addPaths( paths );
+	}
+
+	StringVectorDataPtr addResult = new StringVectorData;
+	StringVectorDataPtr removeResult = new StringVectorData;
+
+	addPaths.paths( addResult->writable() );
+	removePaths.paths( removeResult->writable() );
+
+	sort( addResult->writable().begin(), addResult->writable().end() );
+	sort( removeResult->writable().begin(), removeResult->writable().end() );
+
+	addStringPlug->setValue( addResult );
+	removeStringPlug->setValue( removeResult );
+}
+
+EditScopeAlgo::SetMembership EditScopeAlgo::getSetMembership( Gaffer::EditScope *scope, const std::string &set, const std::string &path )
+{
+	Node *processor = scope->acquireProcessor( "SetMembershipEdits", /* createIfNecessary */ false );
+	if( !processor )
+	{
+		return EditScopeAlgo::SetMembership::INHERIT;
+	}
+
+	auto rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+
+	auto row = rows->row( set );
+	if( !row )
```

Comment:
We should also return `INHERIT` if the row exists but is disabled. Which raises the question, what should `setSetMembership()` do when it finds a disabled row? Perhaps it should enable it, but clear out any previous entries from `add` and `remove`?

Reply:
That sounds reasonable. Fixed in 39c45ac6af6be88fd889900b5785381e31dfb763